File: backend/app/Models/Category.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;

class Category extends Model
{
    use HasFactory;

    protected $fillable = [
        'name',
        'name_ar',
        'slug',
        'icon',
        'type',
        'sort_order',
        'is_active',
    ];

    protected $casts = [
        'is_active' => 'boolean',
        'sort_order' => 'integer',
    ];

    public function channels(): HasMany
    {
        return $this->hasMany(Channel::class)->orderBy('sort_order');
    }

    public function packages(): HasMany
    {
        return $this->hasMany(Package::class)->orderBy('sort_order');
    }
}

File: android/app/src/main/java/com/sportiptv/app/domain/model/Category.kt
package com.sportiptv.app.domain.model

data class Category(
    val id: Long,
    val name: String,
    val nameAr: String,
    val slug: String,
    val icon: String?,
    val type: String,
    val sortOrder: Int,
    val isActive: Boolean,
    val channelsCount: Int
)
